explorer/src/components/SolanabeachComponents/NetworkStats.js

```diff
@@ -0,0 +1,461 @@
+import React, { Fragment, Component } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Row, Col, CardBody, Card, Progress } from "reactstrap";
+import NumberFormat from "react-number-format";
+import BlockHeight from "./BlockHeight";
+import LeaderWorldMap from "./LeaderWorldMap";
+import "moment-duration-format";
+import PerformanceHistory from "./PerformanceHistory";
+
+const moment = require("moment");
+const DROP = 1000000000;
+
+export default class NetworkStats extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      RPCdata: {
+        stakeWarmup: {
+          msUntilWarmup: 0,
+          epochsUntilWarmup: 0,
+          warmedUpStakePercentage: 0,
+        },
+      },
+    };
+  }
+
+  componentDidMount() {
+    const { socket } = this.props;
+    socket.on("dashboardInfo", (RPCdata) => this.handleRPCdata(RPCdata));
+  }
+
+  handleRPCdata(data) {
+    var RPCdata = {};
+
+    //TODO: find out why backend reports data.epochInfo.slotsInEpoch as 0 from time to time
+    // temp fix:
+    let slotsInEpoch;
+    if (data.epochInfo.slotsInEpoch === 0) {
+      slotsInEpoch = 420000;
+    } else {
+      slotsInEpoch = data.epochInfo.slotsInEpoch;
+    }
+
+    // Block Time & Epoch & Skip Rate
+    RPCdata.avgBlockTime_1min = data.avgBlockTime_1min;
+    RPCdata.avgBlockTime_1h = data.avgBlockTime_1h;
+    RPCdata.epochPercent = (data.epochInfo.slotIndex / slotsInEpoch) * 100;
+    RPCdata.msUntilNextEpoch =
+      (slotsInEpoch - data.epochInfo.slotIndex) *
+      RPCdata.avgBlockTime_1h *
+      1000;
```

Comment:
Again, Solana Beach isn't correct here. It calculates the average block time of all completed slots and ignores skipped slots. So it thinks that the slot time is slower than it actually is and so this time remaining is pretty far off. You can check out my calculation here: https://github.com/solana-labs/solana/blob/97c0f7c3646a5f07dbdb424aeb17a33e9390b7e9/explorer/src/utils/index.ts#L7-L11